### PR TITLE
Add ISCE3

### PR DESCRIPTION
**Pull request by rtburns-jpl**
Adds ISCE<sup>3</sup>, a library and scientific computing environment for processing [InSAR (Interferometric Sythetic-Aperture Radar)](https://en.wikipedia.org/wiki/Interferometric_synthetic-aperture_radar) data.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rtburns-jpl** [comment]
<details>
<details>
<summary>Spoiler</summary>
<details>
<summary>check 'em</summary>

</details>
</details>
</details>

File: recipes/isce3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    - cython
+    - gtest
+    - ninja
+  host:
+    - cereal
+    - eigen
+    - fftw
+    - gdal
+    - h5py
+    - numpy {{ numpy }}
```
Comment (chrisburr) [inline recipes/isce3/meta.yaml:33]: If this is just using pybind11 for numpy support, then you might not need numpy in `host` at all.

If you're using the C API directly, see https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

Regardless the `{{ numpy }}` can be removed.

File: recipes/isce3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    - pybind11
+    - python
+    - pip
+    - pyre
+    - pytest
+    - ruamel.yaml
+    - yamale
```
Comment (chrisburr) [inline recipes/isce3/meta.yaml:40]: Have you checked if these are all needed in `host`. They might only be needed in `run`.

File: recipes/isce3/meta.yaml (hunk 1 above)
Comment (rtburns-jpl) [reply recipes/isce3/meta.yaml:33]: Numpy support is needed for some remaining cython components (which will eventually be phased out in favor of pybind11). So I think pin_compatible is the right mechanism, if I understand correctly

File: recipes/isce3/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    - pyre
+    - python
+    - ruamel.yaml
+    - yamale
+
+test:
+  requires:
+    - pytest
+  commands:
+    - cd build && ctest --output-on-failure
```
Comment (chrisburr) [inline recipes/isce3/meta.yaml:55]: Apologies, you need to move these back as they're not actually working:
```
+ cd build
...run_test.sh: line 7: cd: build: No such file or directory
```
(I'm a little surprised the CI doesn't fail...)

This step should be running tests on the installed files so the `build` directory doesn't exist any more. There is a way to keep it around, but it's risky as many packages accidentally pick up a dependency on the build area. The best solution is to run `ctest` during the build.

Comment (rtburns-jpl) [reply recipes/isce3/meta.yaml:55]: Ah, thanks. It looks like having pytest in `test: requires:` doesn't work with that though, so I'd have to move it back to host requirements.

Comment (rtburns-jpl) [reply recipes/isce3/meta.yaml:55]: Hmm, actually all of those python package run requirements are needed during build if we're running the tests in build.sh. How can I add them back without messing up the dependencies?

Comment (chrisburr) [reply recipes/isce3/meta.yaml:55]: In that case you can add them back (maybe with a comment saying there needed for testing). Sorry for sending you around in circles! 😄

Comment (rtburns-jpl) [reply recipes/isce3/meta.yaml]: No worries, thanks for helping out. I'm pretty familiar with a cross-compiling build/host/target split but wasn't too clear on conda's host/run, so I think a bit of trial and error has helped me to understand it much better :)

**Pull request merged**